packages/messenger/Tests/Broker/Command/StartMessengerBrokerCommandTest.php

```diff
@@ -148,4 +214,103 @@ function (BrokerStartedEvent $event) use ($concurrent, $timeout, $context): void
             )
         );
     }
+
+    /**
+     * @dataProvider provideDataForTestExecuteWithAutoConcurrent
+     */
+    public function testExecuteWithAutoConcurrent(
+        int $numCpus,
+        float $processesPerCore,
+        int $minProcesses,
+        int $concurrent
+    ): void {
+        $this
+            ->cpuCounter
+            ->method('count')
+            ->willReturn($numCpus);
+
+        $timeout = random_int(1, 10);
+        $context = uniqid('context-');
+
+        $this->eventDispatcher->addListener(
+            BrokerStartedEvent::class,
+            function (BrokerStartedEvent $event) use ($concurrent, $timeout, $context): void {
+                $this->assertSame(
+                    $context,
+                    $event->getBroker()->getContext()
+                );
+
+                $this->assertSame(
+                    $concurrent,
+                    $event->getConcurrent()
+                );
+
+                $this->assertSame(
+                    $timeout,
+                    $event->getTimeout()
+                );
+
+                $broker = $event->getBroker();
+
+                $this->assertSame(
+                    $this->processFactory,
+                    ReflectionAccessor::getPropertyValue($broker, 'processFactory')
+                );
+
+                $this->assertSame(
+                    $this->eventDispatcher,
+                    ReflectionAccessor::getPropertyValue($broker, 'eventDispatcher')
+                );
+
+                $this->assertSame(
+                    $this->consolePath,
+                    ReflectionAccessor::getPropertyValue($broker, 'consolePath')
+                );
+
+                $broker->stop();
+            }
+        );
+
+        $this
+            ->execute([
+                '--context' => $context,
+                '--concurrent' => 'auto',
+                '--timeout' => $timeout,
+                '--processes-per-core' => $processesPerCore,
+                '--minimum-processes' => $minProcesses,
+            ])
+            ->test(CommandDataTester::create(
+                0,
+                [
+                    '[OK] Broker starting.',
+                    sprintf('! [NOTE] Concurrency %d', $concurrent),
+                    sprintf('! [NOTE] Timeout %d', $timeout),
+                    '[OK] Broker stopped. ',
+                ]
+            ));
+    }
+
+    public static function provideDataForTestExecuteWithAutoConcurrent(): iterable
+    {
+        yield [
```

Comment:
add a 'key' to describe what you are testing.

yield 'integer multiplier' =>
yield 'float multiplier' =>
yield 'minimum process' =>